core/src/main/java/org/apache/carbondata/core/scan/filter/executer/RowLevelFilterExecutorImpl.java

```diff
@@ -106,6 +108,11 @@
    */
   boolean isNaturalSorted;
 
+  /**
+   * date direct dictionary generator
+   */
+  private DirectDictionaryGenerator dateDictionaryGenerator;
+
   public RowLevelFilterExecutorImpl(List<DimColumnResolvedFilterInfo> dimColEvaluatorInfoList,
```

Comment:
better to use IncludeFilterExecutorImpl as much as possible.

Reply:
Have checked this. We are using IncludeFilterExecutorImpl as long as their is no cast expression involved. Cast operation is involved only when the data type of column to compare is  not same as the literal values it is compared with.
      <p>Each literal value in the InFilter can be of different type. Spark chooses one datatype for the entire list of literal values.  
For example,
i IN (3, 2, 1.0) -> all values treated as decimal
i IN (3, 2.0, '1.0') -> all values are treated as string. Note: 2.0 is not same as 2 when casted as string.
<p> When the cast operation is present, `SparkUnknownExpression` with spark cast(`SparkExpression`) is used. `SparkUnknownExpression.evaluate()` takes row's column value.  Using spark cast expression, validates it and converts it to datatype of literal value it chose. And then compare the converted column value against literal values. So validation and type conversions are handled with spark cast expression.
IMO, IncludeFilterExecutorImpl is already used when possible.

Reply:
in my opinion, when we need to use RowLevelFilterExecutorImpl,  better to consider why we pushdown this predicate into the create relation at first.